### DISCUSSION TITLE
Reset and Refetch Infinite Query

### DISCUSSION THREAD

**Discussion by arsotech** [TanStack/query#6800] [Q&A]
I have a use case where I pull data via an infinite query. When data becomes stale or mutates, I want to reset the query so I can refetch the 1st page. We don't want to refetch all pages because the data set could be huge. I understand I can do a query refetch but refetch will refetch all pages. And if I set the max pages, it breaks the pages and limits the new data set to the number of max pages which is counter intuitive. My solution is I do resetQueries. However, that clears the data to initial state so my component clears. From a user experience, we should be resetting or refetching (kind of in the background) and the reset happens when the new data comes through. Any thoughts? Thanks.

**Comment by TkDodo** [comment]
call `setQueryData` and strip everything except the first page

**Reply by arsotech** [reply]
That worked like a charm. Thanks.

**Comment by RamNikhil411** [comment]
use placeholderData and reset it will work fine
